rotkehlchen/api/v1/schemas.py

```diff
@@ -1027,72 +1018,111 @@ def make_history_base_entry(self, data: dict[str, Any], **_kwargs: Any) -> dict[
-                counterparty=(counterparty := data['counterparty']),
-                product=(product := data['product']),
-                address=(address := data['address']),
-            ), EvmSwapEvent(
-                tx_hash=tx_hash,
-                sequence_index=sequence_index + 1,
-                timestamp=timestamp,
-                location=location,
-                event_subtype=HistoryEventSubType.RECEIVE,
-                asset=data['receive_asset'],
-                amount=data['receive_amount'],
-                location_label=location_label,
-                notes=receive_notes,
-                identifier=context_schema.get_grouped_event_identifier(
-                    data=data,
-                    subtype=HistoryEventSubType.RECEIVE,
-                    sequence_index_offset=1,
-                ),
-                event_identifier=event_identifier,
-                counterparty=counterparty,
-                product=product,
-                address=address,
-            )]
-            if (fee_asset := data['fee_asset']) is not None:
-                events.append(EvmSwapEvent(
-                    tx_hash=tx_hash,
-                    sequence_index=sequence_index + 2,
-                    timestamp=timestamp,
-                    location=location,
-                    event_subtype=HistoryEventSubType.FEE,
-                    asset=fee_asset,
-                    amount=data['fee_amount'],
-                    location_label=location_label,
-                    notes=fee_notes,
-                    identifier=context_schema.get_grouped_event_identifier(
-                        data=data,
-                        subtype=HistoryEventSubType.FEE,
-                        sequence_index_offset=2,
-                    ),
-                    event_identifier=event_identifier,
-                    counterparty=counterparty,
-                    product=product,
-                    address=address,
-                ))
-
-            return {'events': events}
+            tx_hash = data['tx_hash']
+            sequence_index = data['sequence_index']
+            timestamp = data['timestamp']
+            location = data['location']
+            event_identifier = data['event_identifier']
+            counterparty = data['counterparty']
+            product = data['product']
+            address = data['address']
+            extra_data = data['extra_data']
+            identifiers = data.get('identifiers', {})
+            location_labels = data.get('location_labels', {})
+            user_notes = data.get('user_notes', {})
+
+            events, is_multi = [], False
+            for subtype in (
+                HistoryEventSubType.SPEND,
+                HistoryEventSubType.RECEIVE,
+                HistoryEventSubType.FEE,
+            ):
+                subtype_identifiers = identifiers.get(subtype_str := subtype.serialize(), [])
+                try:
+                    amounts_and_assets = list(zip(
```

Comment:
is `list()` needed here? i guess you can enumerate over a zip since it's an iterable